### DISCUSSION TITLE
Could not find valid docker environment on Unix/Azure when upgrading from 1.17.6 to 1.18.x

### DISCUSSION THREAD

**Discussion by poorSock** [testcontainers/testcontainers-java#7051] [Q&A] [answered]
Hi,

I am a tester and using an azure pipeline to run tests. A selenium container gets started for that purpose.

Everything works fine with version 1.17.6
When upgrading to 1.18.0 or 1.18.1 I get following errors, no other changes on my codebase or the environment were made

docker daemon & client API versions seem to be 1.4.2

08:07:13.996 DEBUG TestcontainersConfiguration.readProperties() - Testcontainers configuration overrides will be loaded from file:/home/AzDevOps_azpcontainer/.testcontainers.properties
08:07:14.007 WARN  TestcontainersConfiguration.readProperties() - Attempted to read Testcontainers configuration file at file:/home/AzDevOps_azpcontainer/.testcontainers.properties but the file was not found. Exception message: FileNotFoundException: /home/AzDevOps_azpcontainer/.testcontainers.properties (No such file or directory)
08:07:14.036 INFO  ImageNameSubstitutor.instance() - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
08:07:14.214 DEBUG RootlessDockerClientProviderStrategy.tryEnv() - $XDG_RUNTIME_DIR is not set.
08:07:14.215 DEBUG RootlessDockerClientProviderStrategy.tryFolder() - '/home/AzDevOps_azpcontainer/.docker/run' does not exist.
08:07:14.258 DEBUG RootlessDockerClientProviderStrategy.tryFolder() - '/run/user/1002' does not exist.
08:07:14.259 DEBUG DockerClientProviderStrategy.tryOutStrategy() - Trying out strategy: UnixSocketClientProviderStrategy
08:07:14.383 DEBUG DockerClientProviderStrategy.tryOutStrategy() - UnixSocketClientProviderStrategy: failed with exception NoClassDefFoundError (com/github/dockerjava/api/command/LoadImageAsyncCmd). Root cause ClassNotFoundException (com.github.dockerjava.api.command.LoadImageAsyncCmd)
08:07:14.393 INFO  DockerMachineClientProviderStrategy.isApplicable() - docker-machine executable was not found on PATH ([/usr/local/openjdk-17/bin, /usr/local/sbin, /usr/local/bin, /usr/sbin, /usr/bin, /sbin, /bin])
08:07:14.394 ERROR DockerClientProviderStrategy.lambda$getFirstValidStrategy$7() - Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
	UnixSocketClientProviderStrategy: failed with exception NoClassDefFoundError (com/github/dockerjava/api/command/LoadImageAsyncCmd). Root cause ClassNotFoundException (com.github.dockerjava.api.command.LoadImageAsyncCmd)As no valid configuration was found, execution cannot continue.
See https://www.testcontainers.org/on_failure.html for more details.

Caused by: java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.lambda$getFirstValidStrategy$7(DockerClientProviderStrategy.java:256)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:247)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:150)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:186)
	at org.testcontainers.DockerClientFactory$1.getDockerClient(DockerClientFactory.java:104)
	at com.github.dockerjava.api.DockerClientDelegate.authConfig(DockerClientDelegate.java:108)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:321)

**Comment by eddumelendez** [comment, marked as answer]
Thanks for the report... are you using docker-java in your build tool? try to use version 3.3.0. I can identify the error in your logs

```
failed with exception NoClassDefFoundError (com/github/dockerjava/api/command/LoadImageAsyncCmd). Root cause ClassNotFoundException (com.github.dockerjava.api.command.LoadImageAsyncCmd)
```

You can run `mvn dependency:tree` or `./gradlew dependencies`

**Reply by poorSock** [reply]
Thanks so much, I guess you meant 3.**3**.0
WebdriverManager still had the old ones
